lib/metadata/wrapper.js

```diff
@@ -67,54 +67,25 @@ const metadata = {
         });
     },
 
-    putObjectMD: (bucketName, objName, objVal, log, cb, params) => {
+    putObjectMD: (bucketName, objName, objVal, params, log, cb) => {
         log.debug('putting object in metdata');
-        client.putObject(bucketName, objName, objVal, log, err => {
-            if (err) {
```

Comment:
These are good arguments, but :
- if we log here, we can be sure all errors are logged. Although the errors should be logged by the calling operation,  it's an easy mistake to omit the logging at that level.
- Although bucketclient logs all the errors, it's helpful from a debugging perspective if these errors are transparent when viewed from S3 logs. That's why it can be a pain when the logs are omitted by accident.
-  OK to have multiple layers of logging.

I don't have too strong an opinion on this one, but it could be worth the trouble to have to update the callback parameters when they are extended.